google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -95,35 +96,61 @@ protected void checkSdk(String path) {
     Set<CloudSdkValidationResult> validationResults =
         CloudSdkService.getInstance().validateCloudSdk(Paths.get(path));
 
-    // if there are any validation errors, show the first one to the user
-    if (!validationResults.isEmpty()) {
-      CloudSdkValidationResult validationResult = validationResults.iterator().next();
-      String message;
-      switch (validationResult) {
-        case CLOUD_SDK_NOT_FOUND:
-          message = appendCloudSdkDownloadMessage(validationResult.getMessage());
-          break;
-        default:
-          message = validationResult.getMessage();
-      }
-      showWarning(message, validationResult.isError());
-    } else {
+    if (validationResults.isEmpty()) {
       hideWarning();
+    } else if (validationResults.size() == 1) {
+      // display the single result
+      CloudSdkValidationResult result = validationResults.iterator().next();
+      showWarning(getMessageForValidationResult(result), result.isError());
+    } else {
+      // use a sorted set to guarantee consistent ordering of CloudSdkValidationResults
+      validationResults = new TreeSet<>(validationResults);
+
+      // display all validation results as a list
+      StringBuilder builder = new StringBuilder();
+      boolean containsErrors = false;
+
+      int counter = 0;
+      for (CloudSdkValidationResult validationResult : validationResults) {
+        if (validationResult.isError()) {
+          containsErrors = true;
+        }
+        if (counter == 0) {
+          builder.append(getMessageForValidationResult(validationResult));
+        } else {
+          builder.append("<p>" + getMessageForValidationResult(validationResult) + "</p>");
+        }
+        counter++;
+      }
+
+      showWarning(builder.toString(), containsErrors);
+    }
+  }
+
+  private String getMessageForValidationResult(CloudSdkValidationResult cloudSdkValidationResult) {
+    String message;
+    switch (cloudSdkValidationResult) {
```

Comment:
I feel like this method and the switch statement make the flow a little less obvious than it could be.

for instance would appendCloudSdkDownloadMessage ever be called with any other validation result other than CLOUD_SDK_NOT_FOUND?